src/components/graph-view.js

```diff
@@ -674,84 +736,133 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
+    const {
+      selectingNode,
+      selectedNodeObj,
+      selectionStart,
+      selectionEnd,
+    } = this.state;
 
     if (this.isPartOfEdge(d3.event.target)) {
       this.handleEdgeSelected(d3.event);
 
       return; // If any part of the edge is clicked, return
     }
 
+    // Check if selecting multiple nodes
+    if (event.shiftKey && event.ctrlKey && selectionStart) {
+      this.handleMultipleSelected(selectionStart, selectionEnd);
+
+      this.setState({
+        selectionStart: null,
+        selectionEnd: null,
+      });
+
+      return;
+    }
+
+    // Check if selecting a node
     if (selectingNode) {
       this.setState({
         focused: true,
         selectingNode: false,
         svgClicked: true,
       });
-    } else {
-      if (
-        !d3.event.shiftKey &&
-        onBackgroundClick &&
-        d3.event.target.classList.contains('background')
-      ) {
-        const xycoords = d3.mouse(d3.event.target);
 
-        onBackgroundClick(xycoords[0], xycoords[1], d3.event);
-      }
+      return;
+    }
 
-      const previousSelection = selectedNodeObj?.node || null;
+    if (
+      !d3.event.shiftKey &&
+      onBackgroundClick &&
+      d3.event.target.classList.contains('background')
+    ) {
+      const xycoords = d3.mouse(d3.event.target);
 
-      // Clicking with ctrl will paste
-      if (
-        onPasteSelected &&
-        !readOnly &&
-        d3.event.ctrlKey &&
-        previousSelection
-      ) {
-        const xycoords = d3.mouse(d3.event.target);
+      onBackgroundClick(xycoords[0], xycoords[1], d3.event);
+    }
 
-        onPasteSelected(previousSelection, xycoords);
+    const previousSelection = selectedNodeObj?.node || null;
 
-        return;
-      }
+    // Clicking with ctrl will paste
+    if (onPasteSelected && !readOnly && d3.event.ctrlKey && previousSelection) {
+      const xycoords = d3.mouse(d3.event.target);
 
-      // de-select the current selection
-      this.setState({
-        selectedNodeObj: null,
-        focused: true,
-        svgClicked: true,
-      });
+      onPasteSelected(previousSelection, xycoords);
 
-      if (onSelectNode) {
-        onSelectNode(null);
-      }
+      return;
+    }
 
-      if (previousSelection) {
-        this.syncRenderNode(previousSelection);
-      }
+    if (!readOnly && d3.event.shiftKey && onCreateNode) {
+      const xycoords = d3.mouse(d3.event.target);
 
-      if (!readOnly && d3.event.shiftKey && onCreateNode) {
-        const xycoords = d3.mouse(d3.event.target);
+      onCreateNode(xycoords[0], xycoords[1], d3.event);
 
-        onCreateNode(xycoords[0], xycoords[1], d3.event);
-      }
+      return;
+    }
+
+    // de-select the current selection
+    this.setState({
+      selectedNodeObj: null,
+      focused: true,
+      svgClicked: true,
+    });
+
+    onSelectNode && onSelectNode(null);
+    onSelectEdge && onSelectEdge(null);
+    onSelectNodes && onSelectNodes(null);
```

Comment:
Maybe we should consider creating single callback for multi-select `onSelect` which passes `{ nodes, edges }`?

Reply:
We can consider that for future updates. We should deprecate the old onSelect for the 8.0 version.